test/watch_test.go

```diff
@@ -0,0 +1,200 @@
+package test
+
+import (
+	"context"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	clientv3 "go.etcd.io/etcd/client/v3"
+)
+
+// TestWatch is unit testing for the Watch operation.
+func TestWatch(t *testing.T) {
+	ctx := context.Background()
+	client := newKine(t)
+
+	var (
+		revAfterCreate int64
+		revAfterUpdate int64
+		revAfterDelete int64
+
+		key          = "testKey"
+		value        = "testValue"
+		updatedValue = "testUpdatedValue"
+	)
+
+	// start watching for events on key
+	watchCh := client.Watch(ctx, key)
+
+	t.Run("ReceiveNothingUntilActivity", func(t *testing.T) {
+		g := NewWithT(t)
+		g.Consistently(watchCh, "100ms").ShouldNot(Receive())
+	})
+
+	t.Run("Create", func(t *testing.T) {
+		g := NewWithT(t)
+
+		// create a key
+		{
+			resp, err := client.Txn(ctx).
+				If(clientv3.Compare(clientv3.ModRevision(key), "=", 0)).
+				Then(clientv3.OpPut(key, value)).
+				Commit()
+
+			g.Expect(err).To(BeNil())
+			g.Expect(resp.Succeeded).To(BeTrue())
+		}
+
+		// receive event
+		t.Run("Receive", func(t *testing.T) {
+			g := NewWithT(t)
+			g.Eventually(watchCh, "50ms").Should(Receive(Satisfy(func(v clientv3.WatchResponse) bool {
```

Comment:
I think "50ms" is a problem. Right now the implementation is polling every 1s, so this test could (should?) fail.